data/transactions/logic/eval.go

```diff
@@ -586,12 +586,16 @@ func (cx *evalContext) step() {
 			if len(spec.Returns) > 1 {
 				num = len(spec.Returns)
 			}
-			if len(cx.stack) < num {
-				cx.err = fmt.Errorf("stack underflow: expected %d, have %d", num, len(cx.stack))
-				return
-			}
-			for i := 1; i <= num; i++ {
-				stackString += fmt.Sprintf("(%s) ", cx.stack[len(cx.stack)-i].String())
+			// check for nil error here, because we might not return
+			// values if we encounter an error in the opcode
+			if cx.err == nil {
```

Comment:
Hm, what do you mean? Without this check we'll return 'stack underflow' for any situation where `cx.Trace != nil` and there is an error encountered in the `asset_params_get` opcode. I don't see the similar check you are referring to.

(Checking for stack underflow when the opcode might not have returned a value seems weird to me... I think the only reason this case isn't triggering for `txn FieldThatDoesNotExist` is that we hit the `len(cx.stack) == 0` special case above.)

Reply:
I meant this block:
```
	for (cx.err == nil) && (cx.pc < len(cx.program)) {
		prevpc := cx.pc
		cost += cx.checkStep()
		if cx.pc <= prevpc {
			err = fmt.Errorf("pc did not advance, stuck at %d", cx.pc)
			return
		}
	}
	if cx.err != nil {
		err = fmt.Errorf("%3d %s", cx.pc, cx.err)
		return
	}
```
here is a similar problem - if ctx.err is not empty but pc is the same, we see **pc did not advance** err instead of actual error. I thought it would be better to address both if you fixed a similar one in `eval`

Reply:
Ah, I understand now. I pushed a fix -- can you please take a look?